### ISSUE TITLE
TypeError: Cannot assign to read only property 'exports' of object '#<Object>'

### ISSUE THREAD

**Issue by felipefadoni** [socketio/socket.io-client#1509]
Hi guys, I updated the latest version of Socket.IO-Client and started giving this error.

Someone knows what it is ?

![Captura de tela de 2021-10-15 23-53-19](https://user-images.githubusercontent.com/6875321/137571083-c93f5176-364a-4f26-b622-5e7943d7a051.png)

```
{
  "name": "aheadpro-ui",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "@types/jest": "^26.0.15",
    "@types/node": "^12.0.0",
    "@types/react": "^17.0.0",
    "@types/react-dom": "^17.0.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "4.0.3",
    "socket.io-client": "^4.3.1",
    "typescript": "^4.1.2",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}

```

**Comment by NeoGlanding** [comment]
i have same problem too.. hope this will be fixed soon...

**Comment by im-on-the-next-level** [comment]
i think that this problem is related [here](https://github.com/socketio/socket.io-client/commit/f0aae8457a8bdf7e2f2286b4b7d34d2798419456#diff-d5a4b3a0309f2337144861084c1014523cced561eb722cb9684c63f5e41e0567). 

when you need quick this problem, remove that `46-48 line` of `/node_modules/socket.io-client/build/esm/index.js`.

![캡처](https://user-images.githubusercontent.com/12643838/137575478-ed3d3d42-e9bd-4687-b08a-9c76b18d1e11.PNG)

**Comment by 8o8inCodes** [comment]
This is so sad

**Comment by 8o8inCodes** [comment]
> Can confirm, issue popped up in `4.3.0`, reverting to `4.2.0` works fine.

You can revert to `4.3.0`, because `4.3.1` is the broken one

**Comment by djaffer** [comment]
this broke our production code a nightmare of the issue to solve. How come there was no testing done.

**Comment by TanmayKumar-EngStud** [comment]
@waakemeup 
try this:- 
first  npm install socket.io-client@4.2.0
in the package.json remove "^" from "^4.2.0" for socket.io-client

**Comment by jagadeesh93** [comment]
@rauchg I have the same issue , I am using  "socket.io-client": "^4.0.1", and  "@types/socket.io-client": "^1.4.36", how can i resolve this issue

**Comment by samuelsogbesan** [comment]
+1, downgrading to 4.3.0 fixed this for me too.

**Comment by darrachequesne** [comment]
This should be fixed in version 4.3.2, sorry for the mess.
